src/main/logutils.ts

```diff
@@ -208,23 +204,19 @@ const determineArenaMatchResult = (line: string): boolean => {
  */
 const handleCombatantInfoLine = (line: string) => {
     const combatantGUID = line.split(',')[1];
+    let combatantInfo = new Combatant(line.split(',')[2]);
 
-    const combatantData: CombatantData = {
-        friendly: false,
-        teamID: line.split(',')[2],
-    }
-
-    combatantInfoMap.set(combatantGUID, combatantData);
+    combatantInfoMap.set(combatantGUID, combatantInfo);
 }
 
 /**
- * Determine if the srcFlags indicate a friendly unit.
+ * Determine if the srcFlags indicate a self unit.
  * @param srcFlags the srcFlags bitmask
- * @returns true if friendly; false otherwise. 
+ * @returns true if self; false otherwise. 
  */
-const isFriendlyUnit = (srcFlags: number): boolean => {
-    const masked = srcFlags & 0x000000f0;
-    return (masked === 0x00000010);
+const isUnitSelf = (srcFlags: number): boolean => {
+    const masked = srcFlags & 0x511;
+    return masked === 0x511;
```

Comment:
Bit nervous about the mind control case, but it seems unlikely that the final event of the game will be from a mind-controlled team mate and trip this up. If it happens we can deal with it when it does!